charts/argocd-image-updater/templates/deployment.yaml

```diff
@@ -84,13 +84,30 @@ spec:
             name: registries-conf
           - mountPath: /tmp
             name: tmp-dir
+          {{- if .Values.authScripts.enabled }}
+          {{- range $k, $v := .Values.authScripts.scripts }}
+          - mountPath: /scripts
+            name: {{ $k | replace "." "-" }}
+          {{- end }}
+          {{- end }}
       volumes:
-      - configMap:
+      - name: registries-conf
+        configMap:
           items:
           - key: registries.conf
             path: registries.conf
           name: argocd-image-updater-config
-        name: registries-conf
+      {{- if .Values.authScripts.enabled }}
+      {{- range $k, $v := .Values.authScripts.scripts }}
+      - name: {{ $k | replace "." "-" }}
+        configMap:
+          defaultMode: 0777
+          items:
+          - key: {{ $k }}
+            path: {{ $k }}
```

Comment:
Why do we need this loop here? What about just adding the configmap directly under /scripts?

Reply:
Wasn't aware that you could mount multiple files from a single configmap, thanks!